vulnerabilities/reentrancy.md

```diff
@@ -73,6 +73,28 @@ function withdraw() external {
 
 In this example, a hacker can exploit this contract by having a fallback function call `transfer()` to transfer spent funds before the balance is set to 0 in the `withdraw()` function.
 
+### Cross-Contract Reentrancy 
+
+Cross-Contract Reentrancy is a sophisticated type of reentrancy attack that leverages the fact of using one contract's state to mainpulate the state of another contract.
+
+```
+// Contract A - Secure Withdraw Function
```

Comment:
@payalkanyan, if you add balances[msg.sender] = 0; to the withdraw function after the funds transfer, then the withdraw function won't be secure anymore.